### PR TITLE
fix script failing on Windows due to '*' in file path

### PR DESCRIPTION
**Pull request by okusha**

### CODE REVIEW

**Comment by okusha** [comment]
I wanted to split this pull request into 2 independent requests but it seems not to be possible.

File: ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsTableModel.java [hunk 1]
```diff
@@ -24,13 +24,12 @@
...
 import net.sourceforge.ganttproject.task.Task;
 
-import javax.swing.table.AbstractTableModel;
 import java.util.*;
 
 /**
  * @author Oleg Kushnikov
  */
-public class ResourceAssignmentsTableModel extends AbstractTableModel {
+public class ResourceAssignmentsTableModel<T> extends TableModelExt {
```
Comment (dbarashev) [inline ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsTableModel.java]: Did you mean to write concrete type, not type variable? `ResourceAssignmentsTableModel extends TableModelExt<ResourceAssignment>` ?

File: ganttproject/src/net/sourceforge/ganttproject/gui/taskproperties/ResourcesTableModel.java [hunk 2]
```diff
@@ -36,7 +36,8 @@
  *
  * @author dbarashev (Dmitry Barashev)
  */
-class ResourcesTableModel extends AbstractTableModel {
+class ResourcesTableModel<T> extends TableModelExt
```
Comment (dbarashev) [inline ganttproject/src/net/sourceforge/ganttproject/gui/taskproperties/ResourcesTableModel.java]: Same question about type variable

File: ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java [hunk 3]
```diff
@@ -218,6 +218,30 @@ public static JPanel createDefaultTableAndActions(JComponent table, JComponent a
...
+      protected void onDeleteEvent() {
+        if (getTable().isEditing() && getTable().getCellEditor() != null) {
+          getTable().getCellEditor().stopCellEditing();
+        }
+        modelExt.delete(getTable().getSelectedRows());
+      }
+
+      @Override
+      protected T getValue(int row) {
+        java.util.List<T> values = modelExt.getAllValues();
```
Comment (dbarashev) [inline ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java]: I think you can use short name `List` here. It is used a few times in this file.

File: ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java [hunk 4]
```diff
@@ -218,6 +218,30 @@ public static JPanel createDefaultTableAndActions(JComponent table, JComponent a
     return result;
   }
 
+  public static <T> JPanel createDefaultTableAndActions(JTable table, TableModelExt modelExt) {
```
Comment (dbarashev) [inline ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java]: And here you probably need to add type variable: `TableModelExt<T>`

**Comment by dbarashev** [comment]
> I wanted to split this pull request into 2 independent requests but it seems not to be possible.
One possible way is to branch from the first commit and pull that branch, so that second commit was left in other branch

Another way is to create new branches and new pull requests :)

Anyway, I cherry-picked the first commit into master.

File: ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java (hunk 3 above)
Comment (okusha) [inline ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java]: sure

File: ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java (hunk 4 above)
Comment (okusha) [inline ganttproject/src/net/sourceforge/ganttproject/gui/AbstractTableAndActionsComponent.java]: right

File: ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsTableModel.java (hunk 1 above)
Comment (okusha) [inline ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsTableModel.java]: Yes, you are right. I'm still getting used to generics

File: ganttproject/src/net/sourceforge/ganttproject/gui/taskproperties/ResourcesTableModel.java (hunk 2 above)
Comment (okusha) [inline ganttproject/src/net/sourceforge/ganttproject/gui/taskproperties/ResourcesTableModel.java]: agree

**Pull request merged**